### PR TITLE
Updating external docs for Phone PIN

### PR DESCRIPTION
**Pull request by dnya3698**

### CODE REVIEW

**Comment by rackernexus** [comment]
Build 17 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-a85434e3c/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 18 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-74302f967/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 19 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-471a55377/docs/cloud-identity/v2/). valid for 14 days

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
+.. _reset-phone-pin:
+
+Reset the phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: PIN

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+   POST /v2.0/users/{userId}/RAX-AUTH/phone-pin/reset
+
+Reset the phone pin for a specific user. Users are not allowed to reset
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: PIN

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+their own Phone PINs by calling this service. If a user tries to
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PINs

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+reset their Phone PIN, the service returns a 403 HTTP response.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN,

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+..  note::
+
+    This service is only available to the following roles:
+
+    ``identity:user-admin``: Can reset the Phone PIN for all of the users on
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    the caller's domain.
+
+    ``identity:user-manage``: Can reset the Phone PIN of users in the caller's
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN

**Comment by rackernexus** [comment]
Build 20 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-adff10511/docs/cloud-identity/v2/). valid for 14 days

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    domain who have the identity:user-manage role or below.
+
+A successful reset phone pin request returns no response body with ``HTTP 204``
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN request

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  405, Invalid Method, "The method specified in the request is not valid for
+  the resource identified in the request URI."
+  409, Conflict, "The request could not be completed due to a conflict with
+  the current state of the target resource."
+  503, Service Fault, "Service is not available."
+
+Request
+-------
+
+The following table shows the header parameters for the reset phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+request:
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 2, 2, 2
+
+    X-Auth-Token, String *(Required)*, A valid authentication token.
+
+The following table shows the request parameters for the reset phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+request:
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 2, 2, 2
+
+    ``only_if_missing`` , Boolean *(Optional)*, "If the value of this parameter
+    is true and the specified user already has a phone pin, then service will
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: phone PIN, then the service

File: api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    return a 409 HTTP response without changing the user's phone pin."
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/reset-phone-pin.rst]: returns

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 12]
```diff
@@ -0,0 +1,95 @@
+.. _unlock-phone-pin:
+
+Unlock the phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+   PUT /v2.0/users/{userId}/RAX-AUTH/phone-pin/unlock
+
+Users can unlock their phone pins by calling this service.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PINs

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+
+
+..  note::
+
+    If users try to unlock a phone pin for any user apart from themselves,
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+    the service returns a ``HTTP 403`` response.
+
+    The service also gives an error with code ``HTTP 403`` if a user attempts to
+    unlock the their phone pin when the phone pin is not in locked state.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN when the phone PIN is...

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+
+The following table shows the possible response codes for this operation:
+
+.. csv-table::
+  :header: Response Code, Name, Description
+  :widths: 2, 2, 2
+
+  204, No Content, "The request has been fulfilled. Phone pin got unlocked."
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: fulfilled. The phone PIN is unlocked."

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+  400, Bad Request, "The request was invalid."
+  401, Unauthorized, "You are not authorized to complete this operation.
+  This error can occur if the request is submitted with an invalid
+  authentication token."
+  403, Forbidden, "Caller does not have an appropriate role or phone pin is"
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN is"  (should there be more text after "is" to complete the sentence?)

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+  404, Not Found, "User not found"
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: "The user was not found."

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+  405, Invalid Method, "The method specified in the request is not valid for
+  the resource identified in the request URI."
+  503, Service Fault, "Service is not available."
+
+Request
+-------
+
+The following table shows the header parameters for the unlock phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+  X-Auth-Token, String *(Required)*, A valid authentication token.
+
+.. note::
+
+    There is no request body for this service.
+
+Response
+--------
+
+A successful unlock phone pin request returns no response body with
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+``HTTP 204`` status code. Response body is only returned in case
+service errors.
+
+**Example: When attempt is made to unlock a phone pin which is
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN that

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+not in locked state JSON response**
+
+.. code::
+
+    {
+        "forbidden": {
+            "code": 403,
+            "message": "User's current phone pin is not in locked state."
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+        }
+    }
+
+**Example: When user without ``phone-pin-admin`` role, tries to
+unlock phone pin for other user JSON response**
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: phone PIN

File: api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+.. code::
+
+    {
+        "forbidden": {
+            "code": 403,
+            "message": "Error code: 'PP-002'; Not Authorized"
+        }
+    }
+
+**Example: When invalid user id is passed in request url JSON response**
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/phone-pin/unlock-phone-pin.rst]: user ID

File: api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst [hunk 25]
```diff
@@ -32,135 +36,96 @@ include the other  body parameters like ``id``, ``enabled``, or
...
        service catalog.
 
+    - Only users with the ``identity:service-admin`` or ``identity:admin``
+      role are allowed to update a user's ``RAX-AUTH:contactId``.
+
+    - Only the ``RAX-AUTH:contactId`` attribute can be updated for a federated
+      user.
+
+    - The ``RAX-AUTH:phonePin`` is only returned if the caller is updating
+      their own user account and a phone pin exists on the account.
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst]: phone PIN

File: api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst [hunk 26]
```diff
@@ -32,135 +36,96 @@ include the other  body parameters like ``id``, ``enabled``, or
...
-|{userId}                  |URI String               |A user ID assigned by    |
-|                          |String *(Required)*      |system when user is      |
-|                          |                         |added.                   |
-+--------------------------+-------------------------+-------------------------+
+.. csv-table::
+   :header: Name, Type, Description
+   :widths: auto
 
+   {userId}, String *(Required)*, "A user ID assigned by system when a user is
+   created"
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst]: created.

File: api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst [hunk 27]
```diff
@@ -32,135 +36,96 @@ include the other  body parameters like ``id``, ``enabled``, or
...
+   information."
+   user.\ **username**, String *(Optional)*, The name to assign to the user.
+   user.\ **email**, String *(Optional)*, Email address for the user account.
+   user.\ **enabled**, String *(Optional)*, "Indicates whether the user is
+   enabled (true) or disabled (false). Users cannot update the ``enabled``
+   status on their own account."
+   user.\ **RAX-AUTH:defaultRegion**, String *(Optional)*, "The default region
+   that the user is assigned to. Must be one of the regions available in the
+   service catalog."
+   user.\ **RAX-AUTH:phonePin**, String *(Optional)*, "Specify a new phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst]: phone PIN

File: api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst [hunk 28]
```diff
@@ -32,135 +36,96 @@ include the other  body parameters like ``id``, ``enabled``, or
...
+   for the user account.  Ensure that the value specified meets the following
+   criteria:
+
+   - Use six numeric digits (such as 871694). A phone pin cannot include more
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst]: phone PIN

File: api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst [hunk 29]
```diff
@@ -32,135 +36,96 @@ include the other  body parameters like ``id``, ``enabled``, or
...
+     than three repeating numbers. (444 is OK, but 4444 is not.) A phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/methods/users/post-user-information-and-password-v2.0.rst]: phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 30]
```diff
@@ -0,0 +1,28 @@
+.. _phone-pin-operations:
+
+=========
+Phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: Phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+=========
+When a customer calls into Rackspace, they will be required to give a 6 digit
+phone pin to the IVR or Support Racker to verify their identity. When a new user
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 32]
```diff
@@ -0,0 +1,28 @@
...
+is created, a 6 digit numeric phone pin is assigned to the user by default. The
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: 6-digit......phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 33]
```diff
@@ -0,0 +1,28 @@
...
+Phone pin cannot be deleted, however a user can update his phone pin
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: phone PIN cannot................his phone PIN by

File: api-docs/api-reference/phone-pin-operations.rst [hunk 34]
```diff
@@ -0,0 +1,28 @@
...
+using ``Update user information and password`` API.
+
+**Update your own phone pin**
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 35]
```diff
@@ -0,0 +1,28 @@
...
+
+.. include:: methods/users/post-user-information-and-password-v2.0.rst
+
+**Retrieve your own phone pin**
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 36]
```diff
@@ -0,0 +1,28 @@
...
+
+.. include:: methods/token/get-validate-token-v2.0-tokens-tokenid.rst
+.. include:: methods/users/get-user-by-id-v2.0.rst
+.. include:: methods/token/post-authenticate-as-tenant-with-token-v2.0-tokens.rst
+
+**Use the following API operation to unlock your phone pin**
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: phone PIN

File: api-docs/api-reference/phone-pin-operations.rst [hunk 37]
```diff
@@ -0,0 +1,28 @@
...
+
+.. include:: methods/phone-pin/unlock-phone-pin.rst
+
+**Use the following API operation to reset the phone pin of another user***
```
Comment (cyrichardson) [inline api-docs/api-reference/phone-pin-operations.rst]: phone PIN

**Comment by rackernexus** [comment]
Build 21 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-da4aca8aa/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 22 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-fcf9df1e8/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 23 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-cd93b61ee/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 24 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-bfd3805ac/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 25 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-86db4cd32/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 26 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-776177c08/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 27 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-2ceec4dea/docs/cloud-identity/v2/). valid for 14 days

**Comment by rackernexus** [comment]
Build 28 was successful. Staging content [Preview!](https://developer-revs.staging.deconst.org/build-518cd771b/docs/cloud-identity/v2/). valid for 14 days

**Pull request merged**